### PR TITLE
Add method for recursively creating a folder

### PR DESCRIPTION
**Pull request by fjvallarino**
@tejohnso @ahmedalsudani please review

### CODE REVIEW

File: platform.js [hunk 1]
```diff
@@ -279,6 +279,18 @@ module.exports = {
...
     });
   },
+  mkdirRecursively(path) {
+    return new Promise((resolve, reject)=>{
+      module.exports.callMkdirp(path, (err)=>{
+        if (!err) {
+          resolve();
+        } else {
+          log.error("Error creating directory", "Error creating directory: " + path);
+          reject({ message: "Error creating directory", userFriendlyMessage: "Error creating directory", error: err });
```
Comment (ahmedalsudani) [inline platform.js]: @fjvallarino it might be useful to append the directory path to the error messages

File: test/unit/platform.js [hunk 2]
```diff
@@ -268,6 +268,24 @@ describe("platform", ()=>{
     });
   });
 
+  it("creates a folder recursively", ()=>{
+    mock(platform, "callMkdirp").callbackWith(null);
+
+    return platform.mkdirRecursively("folder1/folder2").then((err)=>{
+      assert(platform.callMkdirp.called);
```
Comment (ahmedalsudani) [inline test/unit/platform.js]: @fjvallarino could we use https://github.com/jupiter/simple-mock#spylastcall to assert that it's also called with `"folder1/folder1"` as the parameter?

**Comment by ahmedalsudani** [comment]
@fjvallarino looks good :+1: We might want to point the PR at `release-branch` though since that's what we're doing in our other projects

**Comment by fjvallarino** [comment]
@ahmedalsudani I didn't create a release-branch since we are not actually deploying this project independently and it's a new feature, not a change. If I create a branch, we'll have to point rise-launcher-electron to that branch and then switch to master before we release. Not worth it in this case, I think.

**Comment by ahmedalsudani** [comment]
@fjvallarino ah, ok. That makes sense

**Comment by tejohnso** [comment]
:+1:

**Comment by fjvallarino** [comment]
@ahmedalsudani updated and force pushed as a single commit

**Comment by ahmedalsudani** [comment]
@fjvallarino :ok_hand:

**Pull request merged**